packages/@aws-cdk/aws-codebuild/lib/source.ts

```diff
@@ -500,18 +507,20 @@ abstract class ThirdPartyGitSource extends GitSource {
   protected readonly webhookFilters: FilterGroup[];
   private readonly reportBuildStatus: boolean;
   private readonly webhook?: boolean;
+  private readonly webhookTriggersBatchBuild?: boolean;
 
   protected constructor(props: ThirdPartyGitSourceProps) {
     super(props);
 
     this.webhook = props.webhook;
     this.reportBuildStatus = props.reportBuildStatus === undefined ? true : props.reportBuildStatus;
     this.webhookFilters = props.webhookFilters || [];
+    this.webhookTriggersBatchBuild = props.webhookTriggersBatchBuild;
   }
 
   public bind(_scope: CoreConstruct, _project: IProject): SourceConfig {
     const anyFilterGroupsProvided = this.webhookFilters.length > 0;
-    const webhook = this.webhook === undefined ? (anyFilterGroupsProvided ? true : undefined) : this.webhook;
+    const webhook = this.webhook === undefined ? (anyFilterGroupsProvided || this.webhookTriggersBatchBuild ? true : undefined) : this.webhook;
```

Comment:
Like I wrote in https://github.com/aws/aws-cdk/pull/11743#discussion_r548326731 , I'd rather fail if `webhook` is `false`, but `webhookTriggersBatchBuild` is `true`.

Reply:
Ah right. Done. I assumed the api was happy with `webhook` `false` and the other options given it was like that already for filter groups. I added an error for that case now too.